### PR TITLE
gzip compression in Java-Data-SDK on log messages

### PR DESCRIPTION
**Pull request by Avleen-Kathuria**

### CODE REVIEW

**Review by Gaurang-Patel — DISMISSED** [review]

File: data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java [hunk 1]
```diff
@@ -220,6 +228,23 @@ public ApiResponse<String> makeRequest(
...
     }
+
+    if (gZip) {
+      headersParams.put("Content-Encoding", "gzip");
+
+      try {
+
+        byte[] bytes = call.request().body().toString().getBytes(StandardCharsets.UTF_8);
+        ByteArrayOutputStream out = new ByteArrayOutputStream(bytes.length);
+        GZIPOutputStream gzip = new GZIPOutputStream(out);
```
Comment (mukundneharkar) [inline data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java:239]: do we need to close this Stream in the end??

File: data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java [hunk 2]
```diff
@@ -220,6 +228,23 @@ public ApiResponse<String> makeRequest(
...
+        gzip.write(bytes);
+        gzip.flush();
+        gzip.close();
```
Comment (mukundneharkar) [inline data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java:242]: This stream will not be closeed if there is any exception...

**Review by mukundneharkar — CHANGES_REQUESTED** [review]

File: data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java (hunk 1 above)
Comment (Avleen-Kathuria) [reply data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java]: Yes, Done

File: data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java (hunk 2 above)
Comment (Avleen-Kathuria) [reply data-sdk/src/main/java/com/logicmonitor/sdk/data/internal/BatchingCache.java]: Added Finally Block, Done

**Review by mukundneharkar — APPROVED** [review]

**Pull request merged**